src/Serializer/AbstractItemNormalizer.php

```diff
@@ -190,9 +191,14 @@ public function denormalize($data, $class, $format = null, array $context = [])
             if (!$this->serializer instanceof DenormalizerInterface) {
                 throw new LogicException('Cannot denormalize the input because the injected serializer is not a denormalizer');
             }
-            $denormalizedInput = $this->serializer->denormalize($data, $inputClass, $format, $context);
+            try {
+                $denormalizedInput = $this->serializer->denormalize($data, $inputClass, $format, $context);
+            } catch (NotNormalizableValueException $e) {
+                throw new UnexpectedValueException('The input data is misformatted.', $e->getCode(), $e);
+            }
+
             if (!\is_object($denormalizedInput)) {
-                throw new \UnexpectedValueException('Expected denormalized input to be an object.');
+                throw new UnexpectedValueException('Expected denormalized input to be an object.');
```

Comment:
Can we preserve at least the property name containing the bad data. It's already a public information.